File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/CS/Source/Sample/Forms/ClearErrorForm.cs
//----------------------------------------------------------------------------- 
// <copyright file="ClearErrorForm.cs" company="KEYENCE">
//	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
// </copyright>
//----------------------------------------------------------------------------- 
using System;
using System.ComponentModel;
using System.Windows.Forms;

namespace LJV7_DllSampleAll.Forms
{
	public partial class ClearErrorForm : Form
	{
		#region Field
		/// <summary>
		/// Error code of the error to clear
		/// </summary>
		private short _errCode;
		#endregion

		#region Property
		/// <summary>
		/// The error code for the error you wish to clear
		/// </summary>
		public short ErrCode
		{
			get { return _errCode; }
		}
		#endregion

		#region Event
		/// <summary>
		/// Close start event
		/// </summary>
		/// <param name="e"></param>
		protected override void OnClosing(CancelEventArgs e)
		{
			if (DialogResult == DialogResult.OK)
			{
				try
				{
					_errCode = Convert.ToInt16(_txtboxErrCode.Text, 16);
				}
				catch (Exception ex)
				{
					MessageBox.Show(this, ex.Message);
					e.Cancel = true;
					return;
				}
			}

			base.OnClosing(e);
		}
		#endregion

		#region Method
		/// <summary>
		/// Constructor
		/// </summary>
		public ClearErrorForm()
		{
			InitializeComponent();

			// Field initialization
			_errCode = 0;
		}
		#endregion

		private void _btnOk_Click(object sender, EventArgs e)
		{

		}
	}
}

File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/VBNET/Source/Forms/ClearErrorForm.vb
'----------------------------------------------------------------------------- 
' <copyright file="ClearErrorForm.cs" company="KEYENCE">
'	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
' </copyright>
'----------------------------------------------------------------------------- 
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace Forms
	Public Partial Class ClearErrorForm
		Inherits Form
		#Region "Field"
		''' <summary>
		''' Error code of the error to clear
		''' </summary>
		Private _errCode As Short
		#End Region

		#Region "Property"
		''' <summary>
		''' The error code for the error you wish to clear
		''' </summary>
		Public ReadOnly Property ErrCode() As Short
			Get
				Return _errCode
			End Get
		End Property
		#End Region

		#Region "Event"
		''' <summary>
		''' Close start event
		''' </summary>
		''' <param name="e"></param>
		Protected Overrides Sub OnClosing(e As CancelEventArgs)
			If DialogResult = DialogResult.OK Then
				Try
					_errCode = Convert.ToInt16(_txtboxErrCode.Text, 16)
				Catch ex As Exception
					MessageBox.Show(Me, ex.Message)
					e.Cancel = True
					Return
				End Try
			End If

			MyBase.OnClosing(e)
		End Sub
		#End Region

		#Region "Method"
		''' <summary>
		''' Constructor
		''' </summary>
		Public Sub New()
			InitializeComponent()

			' Field initialization
			_errCode = 0
		End Sub
		#End Region

		Private Sub _btnOk_Click(sender As Object, e As EventArgs)

		End Sub
	End Class
End Namespace
